File: src/main/webapp/resources/js/controllers/memberController.js
gymApp.controller('MemberController', ['$http', '$scope', function($http, $scope){

    $scope.findAllMembers = function() {
        $http.get('/api/member/all').success(function (data) {
            $scope.members = data;
        });
    };

    $scope.updateMember = function(member) {
        $http.post('/api/member/update/'+member.memberId, JSON.stringify(member)).success(function () {
            $scope.findAllMembers();
        });
    };
    $scope.createMember = function(member) {
        $http.post('/api/member/create', JSON.stringify(member)).success(function () {
            $scope.clearAdd(member);
            $scope.findAllMembers();
        });
    };
    $scope.deleteMember = function(id) {
        $http.delete('/api/member/delete/'+id).success(function () {
            $scope.findAllMembers();
        });
    };

    $scope.clearAdd = function(member){
        angular.element('#first-name').value = null;
        angular.element('#last-name').value = null;
        angular.element('#phone-number').value = null;

        member.lastName = null;
        member.firstName = null;
        member.phoneNumber = null;
    };

    $scope.showForm = false;
    $scope.findAllMembers();
}]);

File: src/main/java/com/javaclass/gymapp/controller/MemberController.java
package com.javaclass.gymapp.controller;

import com.javaclass.gymapp.domain.Member;
import com.javaclass.gymapp.service.MemberService;
import org.json.JSONObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@Controller
@RequestMapping("/api/member")
public class MemberController {

    @Autowired
    MemberService memberService;

    @ResponseBody
    @RequestMapping(value = "/all", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
    public List<Member> findAllMembers(){
        return memberService.findAll();
    }

    @RequestMapping(value = "/update/{id}", method = RequestMethod.POST)
    public ResponseEntity updateMember(@RequestBody String jsonMember){

        JSONObject object = new JSONObject(jsonMember);

        Member member = new Member();
        member.setMemberId(object.getLong("memberId"));
        member.setFirstName(object.getString("firstName"));
        member.setLastName(object.getString("lastName"));
        try{
            member.setPhoneNumber(object.getString("phoneNumber"));
        }catch (Exception e){

        }

        try {
            memberService.updateMember(member);
            return new ResponseEntity(HttpStatus.OK);
        }catch (Exception e){
            System.out.println(e.getMessage());
            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
        }
    }

    @RequestMapping(value = "/create", method = RequestMethod.POST)
    public ResponseEntity createMember(@RequestBody String jsonMember){

        JSONObject object = new JSONObject(jsonMember);

        Member member = new Member();
        member.setFirstName(object.getString("firstName"));
        member.setLastName(object.getString("lastName"));
        try{
            member.setPhoneNumber(object.getString("phoneNumber"));
        }catch (Exception e){

        }

        try {
            memberService.createMember(member);
            return new ResponseEntity(HttpStatus.OK);
        }catch (Exception e){
            System.out.println(e.getMessage());
            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
        }
    }

    @RequestMapping(value = "/delete/{id}", method = RequestMethod.DELETE)
    public ResponseEntity deleteMember(@PathVariable("id") Long id){
        try {
            memberService.deleteMember(id);
            return new ResponseEntity(HttpStatus.OK);
        }catch (Exception e){
            System.out.println(e.getMessage());
            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
        }
    }
}